fs/azblob/blob_file.py

```
@@ -12,19 +12,35 @@ class BlobFile(io.IOBase):
     """
 
     @classmethod
-    def factory(cls, blob, mode):
+    def factory(cls, blob, mode, version):
         """Create a BlobFile backed with a temporary file."""
         _temp_file = tempfile.TemporaryFile()
-        proxy = cls(_temp_file, blob, mode)
+        proxy = cls(_temp_file, blob, mode, version)
         return proxy
 
     def __repr__(self):
-        return f"BlobFile({self._blob.blob_name}, {self.mode})"
+        return f"BlobFile({self._name()}, {self.mode})"
 
-    def __init__(self, f, blob, mode):
+    def __init__(self, f, blob, mode, version):
         self._f = f
         self._blob = blob
         self.mode = Mode(mode)
+        self.version = version
+
+    def _name(self):
+        if self.version == 1:
+            return self._blob.blob_name
+        if self.version == 2:
+            return self._blob.path_name
+
+    def _upload(self):
+        version = self.version
+        if version == 1:
+            self._blob.upload_blob(self.raw, overwrite=True)
+        elif version == 2:
+            self._blob.upload_data(self.raw, overwrite=True)
+        else:
+            raise ValueError(f"Unknown sdk {version=}")
```

Should we raise the error in the constructor? Right now, the string representation of an object can possibly be made of `None` as returned by the `_name` method if the `version` attribute is neither 1 or 2.